sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClient.java

```diff
@@ -98,37 +104,48 @@
             return Mono.just(requestBuilder.head().build());
         }
 
-        return toOkHttpRequestBody(request.getBody(), request.getHeaders())
+        return toOkHttpRequestBody(request.getContent(), request.getHeaders())
             .map(okhttpRequestBody -> requestBuilder.method(request.getHttpMethod().toString(), okhttpRequestBody)
                 .build());
     }
 
     /**
      * Create a Mono of okhttp3.RequestBody from the given java.nio.ByteBuffer Flux.
      *
-     * @param bbFlux stream of java.nio.ByteBuffer representing request content
+     * @param bodyContent The BinaryData request body
      * @param headers the headers associated with the original request
      * @return the Mono emitting okhttp3.RequestBody
      */
-    private static Mono<RequestBody> toOkHttpRequestBody(Flux<ByteBuffer> bbFlux, HttpHeaders headers) {
-        Mono<okio.ByteString> bsMono = bbFlux == null
-            ? EMPTY_BYTE_STRING_MONO
-            : toByteString(bbFlux);
+    private static Mono<RequestBody> toOkHttpRequestBody(BinaryData bodyContent, HttpHeaders headers) {
+        String contentType = headers.getValue("Content-Type");
+        MediaType mediaType = (contentType == null) ? null : MediaType.parse(contentType);
 
-        return bsMono.map(bs -> {
-            String contentType = headers.getValue("Content-Type");
-            MediaType mediaType = (contentType == null) ? null : MediaType.parse(contentType);
 
-            return RequestBody.create(bs, mediaType);
-        });
+        if (bodyContent == null) {
+            return Mono.defer(() -> Mono.just(RequestBody.create(ByteString.EMPTY, mediaType)));
+        }
+
+        BinaryDataContent content = BinaryDataHelper.getContent(bodyContent);
+        if (content instanceof ByteArrayContent) {
+            return Mono.defer(() -> Mono.just(RequestBody.create(content.toBytes(), mediaType)));
+        } else if (content instanceof FileContent) {
+            FileContent fileContent = (FileContent) content;
+            // This won't be right all the time as we may be sending only a partial view of the file.
+            // TODO (alzimmer): support ranges in FileContent
+            return Mono.defer(() -> Mono.just(RequestBody.create(fileContent.getFile().toFile(), mediaType)));
+        } else if (content instanceof StringContent) {
+            return Mono.defer(() -> Mono.just(RequestBody.create(bodyContent.toString(), mediaType)));
+        } else {
+            return toByteString(bodyContent.toFluxByteBuffer()).map(bs -> RequestBody.create(bs, mediaType));
```

Comment:
is this the best way to handle StreamContent ?

Reply:
Most certainly not, but I also wasn't aiming on changing how it's handled in this PR